src/emmetHelper.ts

```diff
@@ -522,12 +518,19 @@ export function isAbbreviationValid(syntax: string, abbreviation: string): boole
 	return (htmlAbbreviationStartRegex.test(abbreviation) && htmlAbbreviationRegex.test(abbreviation));
 }
 
-function isExpandedTextNoise(syntax: string, abbreviation: string, expandedText: string): boolean {
+function isExpandedTextNoise(syntax: string, abbreviation: string, expandedText: string, options: Partial<Options>): boolean {
 	// Unresolved css abbreviations get expanded to a blank property value
 	// Eg: abc -> abc: ; or abc:d -> abc: d; which is noise if it gets suggested for every word typed
 	if (isStyleSheet(syntax)) {
-		const after = (syntax === 'sass' || syntax === 'stylus') ? '' : ';';
-		return expandedText === `${abbreviation}: \${0}${after}` ||
+		const between = options['stylesheet.between'] ?? ': ';
+		const after = options['stylesheet.after'] ?? ';';
+
+		// Remove overlapping between `abbreviation` and `between`, if any
+		let endPrefixIndex = abbreviation.indexOf(between[0], abbreviation.length - between.length);
```

Comment:
Can you give an example of when this code would apply?

Reply:
Now that the above check is gone, typing `abc:` will generate an emmet completion and this change ensures that it's marked as noise and ignored

Reply:
While checking the `indexOf` there, I noticed that there's `lastIndexOf`. I wonder whether using that would be clearer?

Reply:
yeah also thought about it but `lastIndexOf` will search from back to front and we only need to check the last `between.length` characters, using `lastIndexOf` will check the whole string if there's no match